### DISCUSSION TITLE
*SCAN and iterator as a string - Redis Python library fix

### DISCUSSION THREAD

**Discussion by adulau** [apache/kvrocks#417] [Q&A] [answered]
Hi All,

I patched the official Python Redis library to support the `*SCAN`and iterator scans.

https://github.com/hashlookup/redis-py/commit/e9c53fb1398e636106c0ec0165ba564943564a83

It's not very clear to me who is correct in this. As the Redis protocol returns a string but a standard Redis seems to return a long int as a string.

This fixes the Python Redis while using `*SCAN` on kvrocks back-end. So it works for me ;-)

Cheers

**Comment by ShooterIT** [comment, marked as answer]
Thanks @adulau Redis return the cursor which is `long` type since the cursor is related to dictionary hash slot, but in kvrocks, we store all keys in SST file, we can't find a long type value to record the iterator point, we use the iterated **key** as cursor, so the cursor is `string` type.

**Reply by adulau** [reply]
It makes sense. I'll maintain the fork of the redis-py for some of our projects. I'll extend it a bit for the kvrocks specificity. Thanks for the feedback.

Edit, the [fork is available at the following location](https://github.com/hashlookup/redis-py/commit/e9c53fb1398e636106c0ec0165ba564943564a83).

**Reply by ShooterIT** [reply]
cool, thanks. For some details, we actually have difference with Redis. We also can recommend our users to use your python SDK. Now we support CAS/CAD https://github.com/KvrocksLabs/kvrocks/pull/415, many friends want it, we or you could support them in  your fork of the redis-py.

**Comment by tecbr** [comment]
For future reference only:
This has been fixed with the new configuration ([https://github.com/apache/kvrocks/pull/1489](https://github.com/apache/kvrocks/pull/1489)):
redis-cursor-compatible yes
